lib/facility.rb

```diff
@@ -1,14 +1,74 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees,
+              :daily_hours,
+              :holidays_closed
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @daily_hours = facility_info[:daily_hours]
+    @holidays_closed = facility_info[:holidays_closed]
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      vehicle.registration_date = Date.today
+      reg_fees = {:antique => 25, :ev => 200, :regular => 100} 
+      @collected_fees += reg_fees[vehicle.plate_type]
+      @registered_vehicles << vehicle
+    end
+  end
```

Comment:
Interesting choice here to make a variable on line 32 that holds the amounts of the fees!!! This is actually a common practice, however, instead of storing it in a local variable, you would create a method called `reg_fees` and then just have that hash living in the method, so you could call the method from any method to read the amount of the fee.